packages/web-components/fast-components/src/radio/radio.ts

```diff
@@ -0,0 +1,212 @@
+import { FormAssociated } from "../form-associated";
+import { FASTRadio } from ".";
+
+const radioGroups = new Map<string, FASTRadio[]>();
+
+export class Radio extends FormAssociated<HTMLInputElement> {
+    @attr({ attribute: "readonly", mode: "boolean" })
+    public readOnly: boolean; // Map to proxy element
+    private readOnlyChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.readOnly = this.readOnly;
+        }
+
+        this.readOnly
+            ? this.classList.add("readonly")
+            : this.classList.remove("readonly");
+    }
+
+    @attr
+    public name: string; // Map to proxy element
+    protected nameChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.name = this.name;
+        }
+    }
+
+    /**
+     * The element's value to be included in form submission when checked.
+     * Default to "on" to reach parity with input[type="radio"]
+     */
+    public value: string = "on"; // Map to proxy element.
+    private valueChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.value = this.value;
+        }
+    }
+
+    /**
+     * Provides the default checkedness of the input element
+     * Passed down to proxy
+     */
+    @attr({ attribute: "checked", mode: "boolean" })
+    public checkedAttribute: boolean;
+    private checkedAttributeChanged(): void {
+        this.defaultChecked = this.checkedAttribute;
+    }
+
+    /**
+     * Initialized to the value of the checked attribute. Can be changed independently of the "checked" attribute,
+     * but changing the "checked" attribute always additionally sets this value.
+     */
+    @observable
+    public defaultChecked: boolean = !!this.checkedAttribute;
+    private defaultCheckedChanged(): void {
+        if (!this.dirtyChecked) {
+            // Setting this.checked will cause us to enter a dirty state,
+            // but if we are clean when defaultChecked is changed, we want to stay
+            // in a clean state, so reset this.dirtyChecked
+            this.checked = this.defaultChecked;
+            this.dirtyChecked = false;
+        }
+    }
+
+    /**
+     * The checked state of the control
+     */
+    @observable
+    public checked: boolean = this.defaultChecked;
+    private checkedChanged(): void {
+        if (!this.dirtyChecked) {
+            this.dirtyChecked = true;
+        }
+
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.checked = this.checked;
+        }
+
+        this.dispatchEvent(new CustomEvent("change", { bubbles: true, composed: true }));
+        this.checked ? this.classList.add("checked") : this.classList.remove("checked");
+        this.checkedAttribute = this.checked;
+
+        if (this.checked) {
+            this.updateOtherGroupRadios();
+        }
+    }
+
+    private updateOtherGroupRadios(): void {
+        if (this.name !== undefined && this.parentNode) {
```

Comment:
Is the expectation here that these are direct descendants of a radiogroup?

Reply:
dangit, that was left in before the radioGroups map, not needed, removing.